pandas/tests/frame/test_constructors.py

```diff
@@ -3177,3 +3177,35 @@ def test_frame_allow_non_nano(self, arr):
     def test_frame_from_dict_allow_non_nano(self, arr):
         df = DataFrame({0: arr})
         assert df.dtypes[0] == arr.dtype
+
+
+def test_dtype_warning_on_empty_list_df():
+    # pd.Series([]) without a specified dtype warns the user
+    expected = pd.DataFrame({"a": pd.Series([]), "b": pd.Series([])})
+
+    with tm.assert_produces_warning(FutureWarning):
+        # Lists does not warn the user
+        result = pd.DataFrame({"a": [], "b": []})
```

Comment:
Error
```
E           AssertionError: Did not see expected warning of class 'FutureWarning'

pandas/_testing/_warnings.py:143: AssertionError
```

Reply:
```python
>>> df_with_series = pd.DataFrame({"a": pd.Series([]), "b": pd.Series([])}) # Prints a warning
<stdin>:1: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
>>> df_with_list = pd.DataFrame({"a": [], "b": []}) # Does not print a warning
>>> tm.assert_frame_equal(df_with_series, df_with_list) # Both df are equal
```